sei_dbt/README.md

```diff
@@ -79,17 +108,155 @@ Then, it will:
          path: "path/to/your/database.db"  # Path to your DuckDB database file
          extensions: ['parquet', 'json']   # Optional extensions to load, if needed
             duckdb_project:
+               threads: 4
+         external_root: "data/dev"
 
        test:
          type: duckdb
          path: "test.db"  # Test database path
+         external_root: "data/test" 
          
        prod:
          type: duckdb
          path: "prod.db"  # Production database path
+         external_root: "data/prod"
    ```
+   
+In this `profiles.yml`:
+- We have three environments: `dev`, `test` and `prod`.
+- For each environment, we define:
+  - `external_root`: A variable points to the directory where CSV files or other data are located.  
+  - `dbt-duckdb` only support certain keys in the `profiles.yml`; for example, `external_root`, `settings`, and `config_options`, etc, and each has its own purpose. `custom`, `vars` are not supported. 
+   
+#### Key Usage:
+- **Global Connection Configuration**: You use `profiles.yml` to store environment-specific details such as database credentials, which will be applied to all dbt projects that use this profile.
+- **Sensitive Information**: The `profiles.yml` file is the right place to store sensitive information like database credentials, API tokens, or connection settings. It is generally **not version-controlled**, as it may contain secrets.
+- **Multiple Environments**: You can define multiple environments (like `dev`, `prod`, `test`) within `profiles.yml` and switch between them using the `target` option. This is useful when different teams or environments need to connect to different databases or use different credentials.
+
+#### Default Location:
+- The `profiles.yml` file is located by default in `~/.dbt/profiles.yml`, making it global across multiple dbt projects on your local machine. If you use a Windows machine, it could be in `C:\Users\myusername\.dbt`.
+- You can also override this by explicitly specifying the location of `profiles.yml` with the `DBT_PROFILES_DIR` environment variable.
```

Comment:
In this section, give your recommendation for how it should be set up for this project. Should the profile file be project specific, or should it contain the info for every database you might be accessing stored in a central location? If project specific, how do you make sure that this project is looking at the right yml file?

Reply:
https://docs.getdbt.com/docs/core/connect-data-platform/connection-profiles
from the source you reference, it sounds like it searches the current directory first before going to ~/.dbt/ - so I would suggest defaulting to project specific yaml files like it does there. Then just using `.gitignore` to make sure that the path info in that profile.yml file isn't pushed.

Reply:
> https://docs.getdbt.com/docs/core/connect-data-platform/connection-profiles from the source you reference, it sounds like it searches the current directory first before going to ~/.dbt/ - so I would suggest defaulting to project specific yaml files like it does there. Then just using `.gitignore` to make sure that the path info in that profile.yml file isn't pushed.

Right. The local `profiles.yml` will override the global `profiles.yml`. Either way works for us. A local project-specific `profiles.yml` is better.